cargo install dioxus-cli compilation error

**Problem**

I attempted to build dioxus-cli and it failed with compilation errors.

```
error[E0277]: the trait bound `<... as SendService<...>>::BodyData: Buf` is not satisfied
   --> /home/nmrp3/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/axum-server-0.7.2/src/server.rs:202:87
    |
202 |                         let serve_future = builder.serve_connection_with_upgrades(io, service);
    |                                                    ------------------------------     ^^^^^^^ the trait `Buf` is not implemented for `<... as SendService<...>>::BodyData`
    |                                                    |
    |                                                    required by a bound introduced by this call
    |
    = note: required for `UpgradedSendStreamTask<<... as SendService<...>>::BodyData>` to implement `Future`
    = note: required for `TokioExecutor` to implement `Executor<UpgradedSendStreamTask<...>>`
    = note: required for `TokioExecutor` to implement `Http2UpgradedExec<<... as SendService<...>>::BodyData>`
    = note: required for `TokioExecutor` to implement `Http2ServerConnExec<TowerToHyperServiceFuture<..., ...>, ...>`
    = note: required for `TokioExecutor` to implement `HttpServerConnExec<TowerToHyperServiceFuture<..., ...>, ...>`
note: required by a bound in `hyper_util::server::conn::auto::Builder::<E>::serve_connection_with_upgrades`
   --> /home/nmrp3/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hyper-util-0.1.18/src/server/conn/auto/mod.rs:269:12
    |
257 |     pub fn serve_connection_with_upgrades<I, S, B>(
    |            ------------------------------ required by a bound in this associated function
...
269 |         E: HttpServerConnExec<S::Future, B>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Builder::<E>::serve_connection_with_upgrades`
    = note: the full name for the type has been written to '/tmp/cargo-installiVdlwJ/release/deps/axum_server-cdb25d73a36b03f6.long-type-6092611790418407442.txt'
    = note: consider using `--verbose` to print the full type name to the console
help: consider further restricting the associated type
    |
159 |         A::Future: Send, <<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::BodyData: Buf
    |                          ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

error[E0277]: the trait bound `<... as SendService<...>>::BodyData: Buf` is not satisfied
   --> /home/nmrp3/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/axum-server-0.7.2/src/server.rs:205:25
    |
205 | /                         tokio::select! {
206 | |                             biased;
207 | |                             _ = watcher.wait_graceful_shutdown() => {
208 | |                                 serve_future.as_mut().graceful_shutdown();
...   |
216 | |                             _ = &mut serve_future => (),
217 | |                         }
    | |                         ^
    | |                         |
    | |_________________________the trait `Buf` is not implemented for `<... as SendService<...>>::BodyData`
    |                           required by a bound introduced by this call
    |
    = note: required for `UpgradedSendStreamTask<<... as SendService<...>>::BodyData>` to implement `Future`
    = note: required for `TokioExecutor` to implement `Executor<UpgradedSendStreamTask<...>>`
    = note: required for `TokioExecutor` to implement `Http2UpgradedExec<<... as SendService<...>>::BodyData>`
    = note: required for `TokioExecutor` to implement `Http2ServerConnExec<TowerToHyperServiceFuture<..., ...>, ...>`
    = note: required for `TokioExecutor` to implement `HttpServerConnExec<TowerToHyperServiceFuture<..., ...>, ...>`
    = note: 3 redundant requirements hidden
    = note: required for `&mut Pin<&mut UpgradeableConnection<'_, ..., ..., ...>>` to implement `Future`
    = note: required for `&mut Pin<&mut UpgradeableConnection<'_, ..., ..., ...>>` to implement `IntoFuture`
    = note: the full name for the type has been written to '/tmp/cargo-installiVdlwJ/release/deps/axum_server-cdb25d73a36b03f6.long-type-6092611790418407442.txt'
    = note: consider using `--verbose` to print the full type name to the console
    = note: this error originates in the macro `$crate::select` which comes from the expansion of the macro `tokio::select` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting the associated type
    |
159 |         A::Future: Send, <<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::BodyData: Buf
    |                          ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

error[E0599]: the method `graceful_shutdown` exists for struct `Pin<&mut UpgradeableConnection<'_, TokioIo<...>, ..., ...>>`, but its trait bounds were not satisfied
   --> /home/nmrp3/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/axum-server-0.7.2/src/server.rs:208:55
    |
208 | ...                   serve_future.as_mut().graceful_shutdown();
    |                                             ^^^^^^^^^^^^^^^^^ method cannot be called due to unsatisfied trait bounds
    |
   ::: /home/nmrp3/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hyper-util-0.1.18/src/rt/tokio.rs:75:1
    |
 75 | pub struct TokioExecutor {}
    | ------------------------ doesn't satisfy `_: HttpServerConnExec<TowerToHyperServiceFuture<<<A as Accept<TcpStream, <M as MakeService<SocketAddr, Request<Incoming>>>::Service>>::Service as SendService<Request<Incoming>>>::Service, Request<Incoming>>, <<A as Accept<TcpStream, <M as MakeService<SocketAddr, Request<Incoming>>>::Service>>::Service as SendService<Request<Incoming>>>::Body>`
    |
    = note: the following trait bounds were not satisfied:
            `<<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::BodyData: Buf`
            which is required by `TokioExecutor: HttpServerConnExec<TowerToHyperServiceFuture<<<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::Service, http::Request<hyper::body::Incoming>>, <<A as accept::Accept<tokio::net::TcpStream, <M as MakeService<std::net::SocketAddr, http::Request<hyper::body::Incoming>>>::Service>>::Service as SendService<http::Request<hyper::body::Incoming>>>::Body>`
    = note: the full name for the type has been written to '/tmp/cargo-installiVdlwJ/release/deps/axum_server-cdb25d73a36b03f6.long-type-16492343223015527450.txt'
    = note: consider using `--verbose` to print the full type name to the console

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `axum-server` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `dioxus-cli v0.7.1`, intermediate artifacts can be found at `/tmp/cargo-installiVdlwJ`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

```

**Steps To Reproduce**

in a fish terminal:

```
cargo install dioxus-cli
```

**Expected behavior**

I expected it to build and then install the cli tool.

**Environment:**

```
> rustc --version
rustc 1.92.0-nightly (53a741fc4 2025-10-16)
> uname -a
Linux taiwaif 6.9.7+bpo-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.9.7-1~bpo12+1 (2024-07-03) x86_64 GNU/Linux
```

**Questionnaire**

 I'm interested in fixing this myself but don't know where to start.

It looks like some API version skew at first sight, but I don't know enough about any of the APIs involved. Also, I don't actually know if cargo was building under nightly or release, and I'm not sure how I would check.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -264,7 +264,7 @@ tauri-bundler = { version = "=2.5.0" }
 lru = "0.16.0"
 async-trait = "0.1.88"
 axum = { version = "0.8.4", default-features = false }
-axum-server = { version = "0.7.2", default-features = false }
+axum-server = { version = "0.7.3", default-features = false }
 http-body = { version = "1.0" }
 tower = "0.5.2"
 http = "1.3.1"

